### PR TITLE
Add Module shortcut in Editor

### LINKED ISSUE #1540: Option to have Blocks Extension enabled by default
### What problem does this address?

A user requested it: https://wordpress.org/support/topic/is-it-possible-to-have-the-css-module-box-visible-by-default/

Allowing users to enable blocks extensions by default, as in to show them by default in the Block Sidebar.

I can see some use of it for Blocks CSS/Animations in particular, and it can be added to the Otter Sidebar.

### What is your proposed solution?

Having a toggle in Otter Sidebar that enables this by default. What do you think @Codeinwp/design-team?

___
### [Figma](https://www.figma.com/file/LRrhO1ovZnXFibXVsezR8a/Otter?node-id=3079%3A34986&t=Oj0iMmaD7r8nH9u0-11)

### PR DESCRIPTION
**Pull request by Soare-Robert-Daniel**
Closes #1540 .

### Summary

Added shortcuts for enabling the Module Visibility from the Editor. Instead of clicking on the plus icon for Block Tools, you can enable them to be shown by default. This will save a click for the user.

Things to consider:

- Added a link that points to the Dashboard.
- The tab is opened by default.
- Modules that are disabled from Dashboard will not be present.

### Screenshots 

image

https://github.com/Codeinwp/otter-blocks/assets/17597852/1b47d47d-80e1-4560-a4cb-604b78647607

----

### Test instructions

- Check that the new section is present.
- If you deactivated a Module, that change is reflected in the Dashboard
- No errors, no broken page.
- Sticky does not have the module disabled options, so it will always be present.

⚠️ In this build the console will show some counting (you can see in the video). This will be for detecting loops that can slow down the website. If you see an abnormal growth (going from 1 to 500 in a second or the count does not stop), please report and post the instance.

ℹ️ The console log can be safely removed after the QA test.

---- 

### Checklist before the final review

- [x] Visual elements are not affected by independent changes.
- [x] It is at least compatible with the [minimum WordPress version](https://wordpress.org/plugins/otter-blocks/).
- [x] It loads additional script in frontend only if it is required.
- [x] Does not impact the [Core Web Vitals](https://web.dev/vitals/).
- [x] In case of deprecation, old blocks are safely migrated.
- [x] It is usable in Widgets and FSE.
- [x] Copy/Paste is working if the attributes are modified.
- [x] PR is following [the best practices]()

### CODE REVIEW

**Review by HardeepAsrani — CHANGES_REQUESTED** [review]
I think you didn't understand what was being discussed in the issue. The toggle shouldn't enable/disable the module but give user an option to have any of those 4 block tools enabled by default if enabled.

For example, if the user toggles on the Blocks CSS, the Block CSS option will appear outside the Block Tools toggle.

Let me know if that makes it clear.

File: src/blocks/plugins/sticky/index.js [hunk 1]
```diff
@@ -48,9 +51,9 @@ const StickyExtension = ( el, props ) => {
 					label={ __( 'Transform to Sticky', 'otter-blocks' ) }
```
Comment (HardeepAsrani) [inline src/blocks/plugins/sticky/index.js:51]: We can exclude this from the workflow. Right now, we enable Transform to Stick when you enable the tool, so it doesn't make sense to include this as default. No one wants to make Sticky as default. Plus, if you disable it then the Sticky settings are disabled too.

**Review by HardeepAsrani — CHANGES_REQUESTED** [review]

File: src/blocks/plugins/options/index.js [hunk 2]
```diff
@@ -16,14 +16,18 @@ import {
...
+	ExternalLink,
 	PanelBody,
 	PanelRow,
-	Snackbar
+	Snackbar, Spinner,
+	ToggleControl
 } from '@wordpress/components';
 
 import {
+	select,
```
Comment (HardeepAsrani) [inline src/blocks/plugins/options/index.js:27]: Is this needed?

**Comment by HardeepAsrani** [comment]
Also, I see this when I try to use:

Screenshot 2023-06-12 at 10 04 30 PM

File: src/blocks/plugins/otter-tools-inspector/index.tsx [hunk 3]
```diff
@@ -20,6 +23,32 @@ import {
...
 import './editor.scss';
 import { useInspectorSlot } from '../../components/inspector-slot-fill/index.js';
+import { useSelect } from '@wordpress/data';
+import { openOtterSidebarMenu } from '../../helpers/block-utility';
+
+const FeaturePanel = ({ props }) => {
+
+	// Function that refresh the inspector slot when a store is updated. This is used for the preferences store.
+	const _ = useSelect( ( select ) => {
+		console.count( 'Refresh Inspector' );
```
[the remark this answers is not in the document]
Comment (Soare-Robert-Daniel) [reply src/blocks/plugins/otter-tools-inspector/index.tsx:33]: I would like to keep it for QA Team

**Comment by Soare-Robert-Daniel** [comment]
@HardeepAsrani, I also added an E2E test for it now that we have the infrastructure.

**Review by HardeepAsrani — APPROVED** [review]

**Comment by irinelenache** [comment]
@Soare-Robert-Daniel Tested and everything's fine here 🚀

**Comment by HardeepAsrani** [comment]
@Soare-Robert-Daniel Can you remove debugging stuff so we can merge it?

**Pull request merged**